### PR TITLE
ImageViewer multiplatform using Voyager for navigation

### PR DESCRIPTION
**Pull request by DevSrSouza**
Hi folks, this PRs adds the [Voyager](https://github.com/adrielcafe/voyager) library for navigation in the ImageViwer sample.

The reason is for simplifying animations, support a Screen stack, supporting back press out of the box in Android.

iOS Preview:

https://user-images.githubusercontent.com/29736164/226504634-1280b7b0-4b27-40c5-b1c8-c6e89bcc5740.mp4

### CODE REVIEW

File: experimental/examples/imageviewer/shared/src/commonMain/kotlin/example/imageviewer/view/CompositionLocals.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package example.imageviewer.view
+
+import androidx.compose.runtime.ProvidableCompositionLocal
+import androidx.compose.runtime.staticCompositionLocalOf
+import example.imageviewer.Dependencies
+
+internal val LocalDependencies: ProvidableCompositionLocal<Dependencies> =
+    staticCompositionLocalOf { error("Dependencies not initialized") }
```
Comment (dima-avdeev-jb) [inline experimental/examples/imageviewer/shared/src/commonMain/kotlin/example/imageviewer/view/CompositionLocals.kt:8]: It is an interesting suggestion, but I think this change should be made in another PR.
We can apply this logic by ourselves in another branch with big changes.

**Comment by dima-avdeev-jb** [comment]
@DevSrSouza 
Thanks for this PR!
We are preparing big changes in this sample.
Sorry, but your change for now conflicts with another branch.
We should postpone your changes.

**Comment by DevSrSouza** [comment]
@dima-avdeev-jb 
No problem at all, I will wait and see if this is faceable when the new code got in the master :D

**Comment by dima-avdeev-jb** [comment]
**New updates:**
Voyager is great, but for now we don't want to add the additional navigation library in ImageViewer
We can add Voyager in the future, but for now, we decided to postpone this decision.
